### PR TITLE
tests: add arxiv functional tests

### LINKED ISSUE #127: Add functional tests to arxiv spider
We have to ensure that the arxiv spider works properly.
So there is need of introducing functional tests to it.

### PR DESCRIPTION
**Pull request by spirosdelviniotis**
* Adds functional test for `arxiv`.

Closes #127

Signed-off-by: Spiros Delviniotis <spirosdelviniotis@gmail.com>

### CODE REVIEW

**Comment by spirosdelviniotis** [comment]
Many commits because it's re-based on #129  :smile:

File: tests/functional/arxiv/fixtures/arxiv_smoke_record.json [hunk 1]
```diff
@@ -0,0 +1,59 @@
+[
```
Comment (david-caro) [inline tests/functional/arxiv/fixtures/arxiv_smoke_record.json]: There's a mixture of tabs and spaces in this file, use only one of them (not a breaking change, but annoying).
Probably spaces is fittest, as all the python files around use them already.

Comment (spirosdelviniotis) [inline tests/functional/arxiv/fixtures/arxiv_smoke_record.json]: Now it's all tabs, but if it's better with spaces, I can do it. :smile:

**Comment by spirosdelviniotis** [comment]
Better to merge this PR after #136 and #129

File: tests/functional/arxiv/fixtures/arxiv_smoke_record.json (hunk 1 above)
Comment (spirosdelviniotis) [inline tests/functional/arxiv/fixtures/arxiv_smoke_record.json]: Done! 2 spaces instead of tabs.

File: docker-compose.test.yml [hunk 2]
```diff
@@ -33,6 +33,10 @@ services:
       - scrapyd
       - ftp_server
 
+  functional_arxiv:
+    <<: *service_base
+    command: py.test -vv tests/functional/arxiv/test_arxiv.py
```
Comment (david-caro) [inline docker-compose.test.yml]: It should use the whole dir, not just the file.

Comment (david-caro) [inline docker-compose.test.yml]:
```yaml
functional_arxiv:
  <<: *service_base
  command: something
  links:
    - one
    - two
```

Comment (david-caro) [inline docker-compose.test.yml]: Refactor common stuff to a dummy 'service_base' service.

Comment (david-caro) [inline docker-compose.test.yml]: Trim the links to only the really needed services (ex. celery does not need the ftp_server)

**Comment by david-caro** [comment]
Hey, I seem to be unable to push to your branch here, is the 'allow maintainers to push' checkbox ticked? Just trying to rebase.

**Comment by david-caro** [comment]
Never mind, I had a typo in the branch name xd

**Pull request merged**